model/host/host.go

```diff
@@ -1778,6 +1788,57 @@ func (h *Host) SetInstanceType(instanceType string) error {
 	return nil
 }
 
+func AggregateSpawnhostData() (*SpawnHostUsage, error) {
+	res := []SpawnHostUsage{}
+	pipeline := []bson.M{
+		{"$match": bson.M{
+			StartedByKey: bson.M{"$exists": true, "$ne": ""},
+			StatusKey:    bson.M{"$in": evergreen.UpHostStatus},
+		}},
+		{"$group": bson.M{
+			"_id":         nil,
+			"hosts":       bson.M{"$sum": 1},
+			"unexpirable": bson.M{"$sum": bson.M{"$cond": []interface{}{"$" + NoExpirationKey, 1, 0}}},
+			"users":       bson.M{"$addToSet": "$" + StartedByKey},
+		}},
+		{"$project": bson.M{
+			"_id":                     "0",
+			"total_hosts":             "$hosts",
+			"total_unexpirable_hosts": "$unexpirable",
+			"num_users_with_hosts":    bson.M{"$size": "$users"},
+		}},
+	}
+
+	if err := db.Aggregate(Collection, pipeline, &res); err != nil {
+		return nil, errors.Wrap(err, "error aggregating hosts")
+	}
+	if len(res) == 0 {
+		return nil, errors.New("no host results found")
+	}
+
+	pipeline = []bson.M{
+		{"$group": bson.M{
+			"_id":     nil,
+			"volumes": bson.M{"$sum": 1},
+			"size":    bson.M{"$sum": "$" + VolumeSizeKey},
+			"users":   bson.M{"$addToSet": "$" + VolumeCreatedByKey},
```

Comment:
an alternative could be to group on users and sum the volumes of the individual users in the projections stage.

Reply:
I don't know a lot about aggregations, so is there a benefit to doing it one way or the other? I like that this is the same as the host aggregation, which may be harder to change. I'm not sure though.

Reply:
I think the end result is the same. The only difference I can think of would be if the users array got too large in memory, but I don't think that's likely at our current scale.

Reply:
With regard to the host aggregation, I think it would work as well, but only if the instance type aggregation were separate.